ServiceApp/src/org/droidplanner/services/android/core/gcs/follow/Follow.java

```diff
@@ -103,11 +123,38 @@ public boolean isEnabled() {
         return state == FollowStates.FOLLOW_RUNNING || state == FollowStates.FOLLOW_START;
     }
 
+    public FollowLocationSource getLocationSource() {
+        return mLocationSource;
+    }
+
+    public void setLocationSource(FollowLocationSource source) {
+        if(mLocationSource != source) {
+            switch(source) {
+                case CLIENT_SPECIFIED: {
+                    Timber.d("Switch to client-specified locations");
+                    locationFinder.removeLocationListener(TAG);
+                    locationFinder.disableLocationUpdates();
+                    break;
+                }
+
+                default: {
+                    Timber.d("Switch to internal locations");
+                    locationFinder.addLocationListener(TAG, this);
+                    locationFinder.enableLocationUpdates();
```

Comment:
@kellyschrock looks like `addLocationListener(...)`/`enableLocationUpdates()` (and `removeLocationListener(...)`/`disableLocationUpdates()`) should always be called together. In that case, we can merge the methods together by removing the `[add|remove]LocationListener(...)` and moving their arguments instead to the `[enable|disable]LocationUpdates()`.